tests/rkt_privatenet_test.go

```diff
@@ -0,0 +1,325 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"sync"
+	"testing"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/ThomasRooney/gexpect"
+	"github.com/coreos/rkt/tests/test_netutils"
+)
+
+/*
+ * No private net
+ * ---
+ * Container must have the same network namespace as the host
+ */
+func TestPrivateNetOmittedNetNS(t *testing.T) {
+	testImageArgs := []string{"--exec=/inspect --print-netns"}
+	testImage := patchTestACI("rkt-inspect-networking.aci", testImageArgs...)
+	defer os.Remove(testImage)
+
+	ctx := newRktRunCtx()
+	defer ctx.cleanup()
+	defer ctx.reset()
+
+	cmd := fmt.Sprintf("%s --debug --insecure-skip-verify run --mds-register=false %s", ctx.cmd(), testImage)
+	t.Logf("Command: %v", cmd)
+	child, err := gexpect.Spawn(cmd)
+	if err != nil {
+		t.Fatalf("Cannot exec rkt: %v", err)
+	}
+
+	expectedRegex := `NetNS: (net:\[\d+\])`
+	result, _ := child.ExpectRegexFind(expectedRegex)
+	if len(result) == 0 {
+		t.Fatalf("Expected %q but not found", expectedRegex)
+	}
+
+	ns, err := os.Readlink("/proc/self/ns/net")
+	if err != nil {
+		t.Fatalf("Cannot evaluate NetNS symlink: %v", err)
+	}
+
+	if nsChanged := ns != result[1]; nsChanged {
+		t.Fatalf("container left host netns")
+	}
+
+	err = child.Wait()
+	if err != nil {
+		t.Fatalf("rkt didn't terminate correctly: %v", err)
+	}
+}
+
+/*
+ * No private net
+ * ---
+ * Container launches http server which must be reachable by the host via the
+ * localhost address
+ */
+func TestPrivateNetOmittedConnectivity(t *testing.T) {
+	httpServeAddr := "0.0.0.0:54321"
+	httpGetAddr := "http://127.0.0.1:54321"
+
+	testImageArgs := []string{"--exec=/inspect --serve-http=" + httpServeAddr}
+	testImage := patchTestACI("rkt-inspect-networking.aci", testImageArgs...)
+	defer os.Remove(testImage)
+
+	ctx := newRktRunCtx()
+	defer ctx.cleanup()
+	defer ctx.reset()
+
+	cmd := fmt.Sprintf("%s --debug --insecure-skip-verify run --mds-register=false %s", ctx.cmd(), testImage)
+	t.Logf("Command: %v", cmd)
+	child, err := gexpect.Spawn(cmd)
+	if err != nil {
+		t.Fatalf("Cannot exec rkt: %v", err)
+	}
+
+	// Child opens the server
+	c := make(chan struct{})
+	go func() {
+		err = child.Wait()
+		if err != nil {
+			t.Fatalf("rkt didn't terminate correctly: %v", err)
+		}
+		c <- struct{}{}
+	}()
+
+	// Host connects to the child
+	go func() {
+		expectedRegex := `serving on`
+		result, _ := child.ExpectRegexFind(expectedRegex)
```

Comment:
If we expect simple things I'd prefer to use expectWithOutput(child, expected) than child.ExpectRegexFind because that would give us the rkt output in case of failure.